views/forms/DropdownCheckboxGroup/index.jsx

```diff
@@ -1,89 +1,119 @@
-    setAnchorEl(null)
-
-    const values = getValues()
+  const options = Object.keys(initialValues)
+  const selectedValues = Object.keys(initialValues).filter(k => initialValues[k].value === 1)
+  const [selectedValue, setSelectedValue] = useState(selectedValues)
 
-    await onSubmit(values)
+  const handleChange = (event) => {
+    const {
+      target: { value },
+    } = event;
+    setSelectedValue(value)
+    onChange(label, value)
   }
-
-  const formValues = useWatch({
-    control,
-    name: filterKey,
-  })
-
   return (
-    <>
-      <TextField
-        label={filterKey}
-        id={`${filterKey}`}
-        placeholder="Select"
-        InputLabelProps={{
-          shrink: true,
-          sx: {
-            color: 'grey.A100',
-            '&.Mui-focused': { color: 'grey.A100' },
-            '&:hover': { cursor: 'none' },
-          },
-        }}
-        onClick={handleClick}
-        onFocus={(e) => e.currentTarget.blur()}
-        sx={{
-          borderColor: 'grey.A100',
+    <>      
+      <FormControl sx={{ m: 1, width: 300 }}>
+      <InputLabel id={`multi-chip-label-${label}`}>{label}</InputLabel>
+      <Select
+        labelId={`multi-chip-label-${label}`}
+        id={`multi-chip-${label}`}
+        multiple
+        value={selectedValue}
+        onChange={handleChange}
+        input={<OutlinedInput id={`select-multiple-${label}`} label={label} />}
+        renderValue={(selected) => {
+          const EtcChip = selected.length > 5 ? <Chip key={"etc"} label="..." /> : <></>
+          return <Box sx={{ display: 'flex', flexWrap: 'nowrap', gap: 0.5 }}>
+            {selected.slice(0, 5).map((value) => {
+              return <Chip key={value} label={value} />
+            }).concat([EtcChip])}
+          </Box>
         }}
-        InputProps={{
-          sx: {
-            '&.Mui-focused': { borderColor: 'grey.A100' },
-          },
-          endAdornment: (
-            <InputAdornment position="start">
-              <KeyboardArrowDown />
-            </InputAdornment>
-          ),
-        }}
-      />
-
-      <Menu
-        id={`${filterKey}`}
-        anchorEl={anchorEl}
-        open={open}
-        onClose={handleClose}
-        disableAutoFocus
-        disableAutoFocusItem
+        MenuProps={MenuProps}
       >
-        {options.map((filter) => {
-          const currentValue = +formValues?.[filter.label].value
-
-          return (
-            <ListItem>
-              <ControlledCheckbox
-                control={control}
-                name={`${filterKey}.${filter.label}.value`}
-                label={filter.label}
-                checked={currentValue === 1}
-              />
-            </ListItem>
-          )
+        {options.map((value) => {
+          return <MenuItem
+            key={value}
+            value={value}
+          >
+            {value}
+          </MenuItem>
         })}
-      </Menu>
-    </>
+      </Select>
+    </FormControl>
+  </>
+    // <>
```

Comment:
did you mean to leave this commented code in?

Reply:
Nope! Sorry about that.

Reply:
273ca56